packages/ubuntu_widgets/test/log_view_test.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:ubuntu_widgets/ubuntu_widgets.dart';
+
+void main() {
+  testWidgets('append lines', (tester) async {
+    final log = StreamController<String>(sync: true);
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: LogView(log: log.stream, maxLines: 2),
+        ),
+      ),
+    );
+
+    final textField = tester.widget<TextField>(find.byType(TextField));
+
+    final controller = textField.controller;
+    expect(controller, isNotNull);
+
+    final scrollController = textField.scrollController;
+    expect(scrollController, isNotNull);
+
+    log.add('line 1');
+    await tester.pump();
+    expect(controller!.text, equals('line 1'));
+    expect(scrollController!.position.extentAfter, equals(0.0));
+    expect(scrollController.position.maxScrollExtent, equals(0.0));
+
+    log.add('line 2');
+    await tester.pump();
+    expect(controller.text, equals('line 1\nline 2'));
+    expect(scrollController.position.extentAfter, equals(0.0));
+    expect(scrollController.position.maxScrollExtent, equals(0.0));
+
+    log.add('line 3');
+    await tester.pump();
+    expect(controller.text, equals('line 1\nline 2\nline 3'));
+    expect(scrollController.position.extentAfter, equals(0.0));
+    expect(scrollController.position.maxScrollExtent, greaterThan(0.0));
+  });
```

Comment:
didrocks made an interesting pair of related questions about the user experience with scrolling this widget ( [this](https://github.com/ubuntu/ubuntu-wsl-splash/pull/2#discussion_r782794192) and [this one](https://github.com/ubuntu/ubuntu-wsl-splash/pull/2#discussion_r782795099) ) which I'd like to take action on, but I'd rather do it in a separate PR, since the purpose of this one is just bring the widgets to this project.

Reply:
I had the same thinking that let's bring in the widgets as they are first, and apply any fixes separately to avoid hiding changes to the import commit.